lib/shelley/src/Cardano/Wallet/Shelley/Launch.hs

```diff
@@ -0,0 +1,373 @@
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+
+-- |
+-- Copyright: © 2018-2020 IOHK
+-- License: Apache-2.0
+--
+-- Provides a function to launch cardano-node for /testing/.
+
+module Cardano.Wallet.Shelley.Launch
+    ( -- * Integration Launcher
+      withCardanoNode
+
+    , NetworkConfiguration (..)
+    , nodeSocketOption
+    , networkConfigurationOption
+    , parseGenesisData
+    ) where
+
+import Prelude
+
+
+import Cardano.BM.Data.Severity
+    ( Severity (..) )
+import Cardano.BM.Trace
+    ( Trace )
+import Cardano.CLI
+    ( optionT )
+import Cardano.Config.Shelley.Genesis
+    ( ShelleyGenesis )
+import Cardano.Launcher
+    ( Command (..), StdStream (..), withBackendProcess )
+import Cardano.Wallet.Logging
+    ( trMessageText )
+import Cardano.Wallet.Network.Ports
+    ( getRandomPort )
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( NetworkDiscriminant (..) )
+import Cardano.Wallet.Primitive.Types
+    ( Block (..), GenesisBlockParameters (..), ProtocolMagic (..) )
+import Cardano.Wallet.Shelley
+    ( SomeNetworkDiscriminant (..) )
+import Cardano.Wallet.Shelley.Compatibility
+    ( NodeVersionData
+    , emptyGenesis
+    , fromGenesisData
+    , mainnetBlockchainParameters
+    , mainnetVersionData
+    , testnetVersionData
+    )
+import Cardano.Wallet.Shelley.Transaction
+    ( genesisBlockFromTxOuts )
+import Control.Applicative
+    ( (<|>) )
+import Control.Exception
+    ( bracket, throwIO )
+import Control.Monad
+    ( forM_, unless, when )
+import Control.Monad.Fail
+    ( MonadFail )
+import Control.Monad.Trans.Except
+    ( ExceptT (..) )
+import Data.Aeson
+    ( eitherDecode, toJSON )
+import Data.Proxy
+    ( Proxy (..) )
+import Data.Text
+    ( Text )
+import Data.Time.Clock
+    ( getCurrentTime )
+import GHC.TypeLits
+    ( SomeNat (..), someNatVal )
+import Options.Applicative
+    ( Parser, flag', help, long, metavar )
+import Ouroboros.Consensus.Shelley.Node
+    ( sgNetworkMagic )
+import Ouroboros.Consensus.Shelley.Protocol
+    ( TPraosStandardCrypto )
+import Ouroboros.Network.Magic
+    ( NetworkMagic (..) )
+import Ouroboros.Network.NodeToClient
+    ( NodeToClientVersionData (..), nodeToClientCodecCBORTerm )
+import System.Directory
+    ( copyFile, doesDirectoryExist, removeDirectoryRecursive )
+import System.Environment
+    ( lookupEnv )
+import System.FilePath
+    ( takeFileName, (</>) )
+import System.Info
+    ( os )
+import System.IO.Temp
+    ( createTempDirectory, getCanonicalTemporaryDirectory )
+
+import qualified Data.Aeson as Aeson
+import qualified Data.ByteString.Lazy as BL
+import qualified Data.HashMap.Strict as HM
+import qualified Data.Text as T
+import qualified Data.Yaml as Yaml
+
+data NetworkConfiguration where
+    MainnetConfig
+        :: (SomeNetworkDiscriminant, NodeVersionData)
+        -> NetworkConfiguration
```

Comment:
I wonder if we shouldn't remove anything regarding "Shelley mainnet" at this stage.

Reply:
Good point, probably. Done.